ignite/contrib/metrics/regression/geometric_mean_absolute_error.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import division
+import torch
+from ignite.exceptions import NotComputableError
+from ignite.metrics.metric import Metric
+
+
+class GeometricMeanAbsoluteError(Metric):
+    r"""
+    Calculates the Geometric Mean Absolute Error.
+
+    :math:`\text{GMRAE} = \sqrt[n]{\prod_{j=1}^{\n}{|A_j - P_j|}`
+
+    where, :math:`A_j` is the ground truth and :math:`P_j` is the predicted value.
+
+    More details can be found in `Botchkarev 2018`__.
+
+    - `update` must receive output of the form `(y_pred, y)`.
+    - `y` and `y_pred` must be of same shape.
+
+    __ https://arxiv.org/abs/1809.03006
+    """
+
+    def reset(self):
+        self._product_of_errors = 1.0
+        self._num_examples = 0
+
+    def update(self, output):
+        y_pred, y = output
+        errors = torch.abs(y.view_as(y_pred) - y_pred)
+        self._product_of_errors = self._product_of_errors * torch.prod(errors).item()
```

Comment:
I think we should implement this differently (e.g. with log) in order to underflow when multiply error values ~1e-1, 1e-2.

Reply:
Agree. I have a working local implementation I can push

Reply:
Added implementation using log.